src/array/empty.js

```diff
@@ -0,0 +1,23 @@
+define(['../lang/isArray'], function(isArray) {
+
+    /**
+     * Array empty
+     */
+
+    function empty(array) {
+        var i, len;
+
+        if(!isArray(array)) {
+            return [];
+        }
+
+        for (i = 0, len = array.length; i < len; i++) {
+            array.pop();
+        }
+
```

Comment:
@jdalton I decided against `array.length` for two reasons.
1. It's [slightly slower than Array#pop](http://jsperf.com/array-destroy/164)
2. Works differently. Doing `array.length = 0` does not prevent array[1] returning an actual value and it doesn't allow garbage collector to remove whatever was in the array (since the references are still there).